src/main/java/salesforce/ui/pages/opportunity/OpportunityClassicPage.java

```diff
@@ -154,6 +161,14 @@ public void clickOnNotesAndAttachmentsButton() {
 
     @Override
     public AbstractTaskOpportunity clickAddTask() {
-        return null;
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(addTaskButton));
+        webDriverWait.until(ExpectedConditions.visibilityOf(addTaskButton));
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(editButton));
+        webDriverWait.until(ExpectedConditions.elementToBeClickable(addFile));
+        webDriverWait.until(ExpectedConditions.visibilityOf(addFile));
+        addTaskButton.click();
+        addTaskButton.click();
+        addTaskButton.click();
```

Comment:
is it the correct? and could be better if you have a method to do this action.
For example:
private void clickOnAddTaskButton() {
      webDriverWait.until(ExpectedConditions.visibilityOf(addTaskButton));
        addTaskButton.click();
}
**In that case you are using Single responsability**

Reply:
yes but that button is unavialable. That's why I am using it.

Reply:
ok, but implement this please
private void clickOnAddTaskButton() {
webDriverWait.until(ExpectedConditions.visibilityOf(addTaskButton));
addTaskButton.click();
}

Reply:
that doesn't wok, because it does't find the element

Reply:
I separate it.